source/MaterialXRenderGlsl/TextureBaker.cpp

```diff
@@ -180,11 +282,69 @@ void TextureBaker::writeBakedDocument(NodePtr shader, const FilePath& filename)
     writeToXmlFile(bakedTextureDoc, filename);
 }
 
-FilePath TextureBaker::generateTextureFilename(OutputPtr output)
+FilePath TextureBaker::generateTextureFilename(OutputPtr output, const string& srName, const string& udim)
```

Comment:
I guess the reason I included the udim string was to be consistent about how I handle different udims. (e.g. specular_roughness_1001.hdr vs. specular_roughness_1002.hdr). Would you recommend that I just create a catch-all suffix term that I pass in as an argument?

Reply:
Some form of UDIM distinction is needed here, and although it could be expressed as an integer instead of a string, the approach you've taken here seems fine.